src/java/org/apache/cassandra/cql3/QueryOptions.java

```diff
@@ -506,7 +509,7 @@ public void encode(QueryOptions options, ByteBuf dest, ProtocolVersion version)
             if (flags.contains(Flag.PAGING_STATE))
                 CBUtil.writeValue(options.getPagingState().serialize(version), dest);
             if (flags.contains(Flag.SERIAL_CONSISTENCY))
-                CBUtil.writeConsistencyLevel(options.getSerialConsistency(), dest);
+                CBUtil.writeConsistencyLevel(options.getSerialConsistency(null), dest);
```

Comment:
see the comment above, passing null here is asking for problems unless we check for null in that method

Reply:
It will eventually be checked in `Guardrails.enabled`